src/openms/source/FORMAT/PepXMLFile.cpp

```diff
@@ -1319,31 +1347,71 @@ namespace OpenMS
 
         optionalAttributeAsString_(aa_mod.aminoacid, attributes, "aminoacid");
         aa_mod.terminus = String(attributeAsString_(attributes, "terminus")).toLower();
-        // "y" if protein terminus, "n" if peptide terminus
-        aa_mod.protein_terminus = String(attributeAsString_(attributes, "protein_terminus")).toLower() == "y";
-        if (aa_mod.terminus == "n")
+        protein_terminus_entry = String(attributeAsString_(attributes, "protein_terminus"));
+      }
+
+      //TODO value "nc" for any terminus is actually not supported by us, yet!
+      if (aa_mod.terminus == "nc")
+      {
+        warning(LOAD, "Warning: value 'nc' for aminoacid terminus not supported."
+                      "The modification will be parsed as a non-restricted modification.");
+      }
+
+      // BIG NOTE: According to the pepXML schema specification, protein terminus is either "c" or "n" if set.
+      // BUT: Many tools will put "Y" or "N" there in conjunction with the terminus attribute.
+      // Unfortunately there is an overlap for the letter "n". We will try to handle both based on the case:
+      String entry_lower = protein_terminus_entry;
+      entry_lower = entry_lower.toLower();
+      if (entry_lower == "y")
+      {
+        aa_mod.protein_terminus = true;
```

Comment:
sorry, but all this code does not belong here!

collect all the information you need and refactor it to `aa_mod`'s implemention.
Ideally, make it a constructor and forbid changing properties individually (to maintain invariants).

This is a typical case of 'just add the code inline' which makes it
 - not testable
 - creates huge unmaintainable classes
 - is no fun to refactor at all

Reply:
aa_mod should not be here at all.
It is:
a) abused also for terminal mods
b) should be our own Residuemodification object, such that we do not lookup ever changing "mod-descriptions" dozens of times